__tests__/data/dynamicConfig_input.json

```diff
@@ -0,0 +1,221 @@
+{
```

Comment:
You may have covered some of the cases already but here is a list of minimum test cases:
- "{{ a.b.c || defaultVal }}"
- "  {{ a.b.c   ||   defaultVal   }}  "
- "{{ a.b.c || "  default value  " }}"
- "{{ a.b.c || "default value }}"
- "{{ a.b.c || default value" }}"
- "{{ a.b.c || 1234 }}"
- "{{ a.b.c || true }}"
- "{{ a.b.c || 45.92340 }}"
- "{ key1: { key2: "{{ a.b.c || defaultVal }}" } }"
- "{ key1: { key2: "{{ a.b.c || defaultVal }}", key3: "{{ d.e || defaultVal }}" } }"
- "{ key1: [ "{{ a.b.c || defaultVal }}" ] }"
- "{ key1: [ "{{ a.b.c || defaultVal }}", "{{ d.e || defaultVal }}" ] }"
- "{ key1: [ { key2: { key3: "{{ a.b.c || defaultVal }}" } } ] }"
- "{ key1: [ [ { key2: { key3: "{{ a.b.c || defaultVal }}" } }, {key4: "{{ d.e || defaultVal }}" } ], "defaultVal" ] }"

If it makes sense, combine some of them in the same case.
@ruchiramoitra @abandonned add if I've missed anything.

Reply:
added the suggested test cases